### PR TITLE
Re #68: Reduce chatty logging for 'condition_error' and 'validation_error'

### PR DESCRIPTION
**Pull request by ehuelsmann**
# Description

As per @oliwel's request, change the logging of the validator's "error"
to DEBUG level: it's not really an error, it's a failure of the user to
provide compliant input (which is at *most* an informational message).

At the same time, add logging of condition errors as a TRACE level
message: these are not failures by any measure; they are merely
conditions evaluating to 'false'. When debugging, it might be handy
to be able to check these, but usually this is too chatty for debugging.
Putting the logging in TRACE instead.

At the same time, remove duplicate logging of condition validation results
which go both through direct logging as well as through logging of `condition_error`s.

Addresses some of #68

## Type of change

Improvement of existing feature.

# Checklist:

- [x] My code follows the style guidelines of this project, please see the [contribution guidelines](https://github.com/jonasbn/perl-workflow/blob/master/CONTRIBUTING.md).
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by oliwel — APPROVED** [review]

File: lib/Workflow/State.pm [hunk 1]
```diff
@@ -182,12 +182,8 @@ sub evaluate_action {
                     # without wrapping it...
                     $wf->{'_condition_result_cache'}->{$orig_condition} = 0;
                     if ( !$opposite ) {
-                        $self->log->is_debug
-                            && $self->log->debug("No access to action '$action_name', condition " .
-                             "'$orig_condition' failed because ' . $EVAL_ERROR");
-
                         condition_error "No access to action '$action_name' in ",
-                            "state '$state' because: $EVAL_ERROR";
+                            "state '$state'; condition '$orig_condition' failed because: $EVAL_ERROR";
```
Comment (jonasbn) [inline lib/Workflow/State.pm:186]: Hi @ehuelsmann 

I would rephrase this to:

```suggestion
                            "state '$state'; condition '$orig_condition' failed due to: $EVAL_ERROR";
```

File: lib/Workflow/State.pm [hunk 2]
```diff
@@ -182,12 +182,8 @@ sub evaluate_action {
...
                     } else {
                         $self->log->is_debug
                             && $self->log->debug("opposite condition '$orig_condition' failed because ' . $EVAL_ERROR");
```
Comment (jonasbn) [inline lib/Workflow/State.pm:189]: Same as above, minor rephrasing

```suggestion
                            && $self->log->debug("opposite condition '$orig_condition' failed due to ' . $EVAL_ERROR");
```

**Review by jonasbn — CHANGES_REQUESTED** [review]
Minor phrasing issues

**Comment by ehuelsmann** [comment]
@jonasbn due to the fact that I incorrectly batched the last two commits, you may want to squash the changes on commit...

**Review by jonasbn — APPROVED** [review]
Looks good
